jmxtrans-cluster/src/main/java/com/googlecode/jmxtrans/cluster/ClusterServiceModule.java

```diff
@@ -0,0 +1,47 @@
+package com.googlecode.jmxtrans.cluster;
+
+import com.google.inject.AbstractModule;
+import com.google.inject.Provides;
+import org.apache.commons.configuration.Configuration;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import static com.google.common.base.Preconditions.checkNotNull;
+
+/**
+ * Created by kulcsart on 5/14/2016.
+ *
+ */
+public class ClusterServiceModule extends AbstractModule {
+    private static final Logger log = LoggerFactory.getLogger(ClusterServiceModule.class);
+    Class implClass;
+    Configuration configuration;
+
+    public ClusterServiceModule(Configuration configuration){
+        this.configuration = configuration;
+
+        checkNotNull(configuration);
+        String className = configuration.getString("provider.classname");
+        checkNotNull(className);
+
+        implClass = getImplClassForName(className);
+        checkNotNull(implClass);
+    }
+
+    @Override
+    protected void configure() {
+        bind(ClusterService.class).to(implClass);
+    }
+
+    @Provides
+    Configuration clusterConfiguration(){return this.configuration;}
+
+    private Class getImplClassForName(String className){
+        try{
```

Comment:
This check logic is probably too complex. Throwing a `ClassNotFoundException` is fine and more explicit than throwing a `NullPointerException`. You should also not log and throw exception. If you throw an exception, it will be catched and logged by the caller (or at least it should), so there is no point in logging it here. What you should do is make sure that the exception has a good message, so that the caller can log it properly.